pkg/oauthserver/oauthserver/oauth_apiserver.go

```diff
@@ -226,38 +229,48 @@ func (c *OAuthServerConfig) buildHandlerChainForOAuth(startingHandler http.Handl
 // TODO TODO, this actually looks a lot like a controller or an add-on manager style thing.  Seems like we'd want to do this outside
 // EnsureBootstrapOAuthClients creates or updates the bootstrap oauth clients that openshift relies upon.
 func (c *OAuthServerConfig) EnsureBootstrapOAuthClients(context genericapiserver.PostStartHookContext) error {
-	webConsoleClient := oauthapi.OAuthClient{
-		ObjectMeta:            metav1.ObjectMeta{Name: OpenShiftWebConsoleClientID},
-		Secret:                "",
-		RespondWithChallenges: false,
-		RedirectURIs:          c.ExtraOAuthConfig.AssetPublicAddresses,
-		GrantMethod:           oauthapi.GrantHandlerAuto,
-	}
-	if err := ensureOAuthClient(webConsoleClient, c.ExtraOAuthConfig.OAuthClientClient, true, false); err != nil {
-		return err
-	}
-
-	browserClient := oauthapi.OAuthClient{
-		ObjectMeta:            metav1.ObjectMeta{Name: OpenShiftBrowserClientID},
-		Secret:                uuid.New(),
-		RespondWithChallenges: false,
-		RedirectURIs:          []string{oauthutil.OpenShiftOAuthTokenDisplayURL(c.ExtraOAuthConfig.Options.MasterPublicURL)},
-		GrantMethod:           oauthapi.GrantHandlerAuto,
-	}
-	if err := ensureOAuthClient(browserClient, c.ExtraOAuthConfig.OAuthClientClient, true, true); err != nil {
-		return err
-	}
-
-	cliClient := oauthapi.OAuthClient{
-		ObjectMeta:            metav1.ObjectMeta{Name: OpenShiftCLIClientID},
-		Secret:                "",
-		RespondWithChallenges: true,
-		RedirectURIs:          []string{oauthutil.OpenShiftOAuthTokenImplicitURL(c.ExtraOAuthConfig.Options.MasterPublicURL)},
-		GrantMethod:           oauthapi.GrantHandlerAuto,
-	}
-	if err := ensureOAuthClient(cliClient, c.ExtraOAuthConfig.OAuthClientClient, false, false); err != nil {
-		return err
-	}
+	// the TODO above still applies, but this makes it possible for this poststarthook to do its job with a split kubeapiserver and not run forever
+	go func() {
```

Comment:
now with the go routine loop it is no `EnsureSomething` anymore, but rather a `StartOAuthClientsBootstrapping`.

Reply:
no need for the error as result.

Reply:
> no need for the error as result.

post start hook.  Don't get a choice